packages/wallets/particlenetwork/src/adapter.ts

```diff
@@ -0,0 +1,178 @@
+
+    private _connecting: boolean;
+    private _wallet: ParticleNetwork | null;
+    private _publicKey: PublicKey | null;
+    private _config: ParticleNetworkAdapterConfig;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+    constructor(config: ParticleNetworkAdapterConfig) {
+        super();
+        this._connecting = false;
+        this._publicKey = null;
+        this._wallet = null;
+        this._config = config;
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.auth.isLogin();
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let wallet: ParticleNetwork;
+            try {
+                wallet = window.particle || new ParticleNetwork(this._config);
```

Comment:
Since this requires instantiating `ParticleNetwork` through a loaded SDK, the loading needs to be done async in here. See how the Torus adapter does this: https://github.com/solana-labs/wallet-adapter/blob/e230958570f192786a27c57d5b39037b898e69dd/packages/wallets/torus/src/adapter.ts#L72-L86